src/gamecontrol.h

```diff
@@ -10,13 +10,102 @@
 #ifndef GAMECONTROL_H
 #define GAMECONTROL_H
 
-void StartNewGame();
-void ShutdownGame();
-
 void OnNewDay();
 void OnNewMonth();
 void OnNewYear();
 void OnNewFrame(uint32 frame_delay);
 
-#endif
+/** Actions that can be run to control the game. */
+enum GameControlAction {
+	GCA_NONE,      ///< No action to run.
+	GCA_NEW_GAME,  ///< Prepare a new game.
+	GCA_LOAD_GAME, ///< Load a saved game.
+	GCA_SAVE_GAME, ///< Save the current game.
+	GCA_QUIT,      ///< Quit the game.
+};
+
+/**
+ * Class controlling the current game.
+ */
+class GameControl {
+public:
+	GameControl();
+	~GameControl();
+
+	void RunAction();
+	void NewGame();
+	void LoadGame(const char *fname);
+	void SaveGame(const char *fname);
+	void QuitGame();
+
+	bool running;
+	GameControlAction next_action; ///< Action game control wants to run, or #GCA_NONE for 'no action'.
 
+private:
+	void NewLevel();
+	void StartLevel();
+	void ShutdownLevel();
+
+	const char *fname;    ///< Filename of game level to load from or save to.
+};
+
+extern GameControl _game_control;
+
+/**
+ * The current game mode controls what user operations that are allowed
+ * and not. In Game mode most construction operations are limited to
+ * owned land.
+ */
+enum GameMode {
+	GM_NONE,   ///< Neither a running game nor in editor. (eg. startup/quit)
+	GM_PLAY,   ///< The current scenario is being played.
+	GM_EDITOR, ///< The current scenario is being edited.
+};
+
+/** Class managing the game mode of the program.
+ *  @todo Move functionality to GameControl class?
+ */
+class GameModeManager {
+public:
+	GameModeManager();
+	~GameModeManager();
+
+	void SetGameMode(GameMode new_mode);
+	inline GameMode GetGameMode() const;
+	inline bool InPlayMode() const;
+	inline bool InEditorMode() const;
+
+private:
+	GameMode game_mode; ///< Current game mode of the program.
+};
```

Comment:
:smile: omg, this looks so unbelievably bloated, an entire class for one boolean :D

Not your current problem though, it's safe to ignore this comment :)